pkg/resources/schema.go

```diff
@@ -208,94 +253,139 @@ func CreateContextSchema(ctx context.Context, d *schema.ResourceData, meta any)
 		PipeExecutionPaused:                     GetConfigPropertyAsPointerAllowingZeroValue[bool](d, "pipe_execution_paused"),
 		Comment:                                 GetConfigPropertyAsPointerAllowingZeroValue[string](d, "comment"),
 	}
+	if v := d.Get("is_transient").(string); v != BooleanDefault {
+		parsed, err := booleanStringToBool(v)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		opts.Transient = sdk.Bool(parsed)
+	}
+	if v := d.Get("with_managed_access").(string); v != BooleanDefault {
+		parsed, err := booleanStringToBool(v)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		opts.WithManagedAccess = sdk.Bool(parsed)
+	}
 	if strings.EqualFold(strings.TrimSpace(name), "PUBLIC") {
 		opts.OrReplace = sdk.Pointer(true)
 	}
 	if err := client.Schemas.Create(ctx, id, opts); err != nil {
-		return diag.FromErr(err)
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Failed to create schema.",
+				Detail:   fmt.Sprintf("schema name: %s, err: %s", id.FullyQualifiedName(), err),
+			},
+		}
 	}
 	d.SetId(helpers.EncodeSnowflakeID(database, name))
 
-	return ReadContextSchema(ctx, d, meta)
+	return ReadContextSchema(false)(ctx, d, meta)
 }
 
-func ReadContextSchema(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
-	client := meta.(*provider.Context).Client
-	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.DatabaseObjectIdentifier)
+func ReadContextSchema(withExternalChangesMarking bool) schema.ReadContextFunc {
+	return func(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+		client := meta.(*provider.Context).Client
+		id := helpers.DecodeSnowflakeID(d.Id()).(sdk.DatabaseObjectIdentifier)
 
-	_, err := client.Databases.ShowByID(ctx, id.DatabaseId())
-	if err != nil {
-		d.SetId("")
-	}
-
-	schema, err := client.Schemas.ShowByID(ctx, id)
-	if err != nil {
-		if errors.Is(err, sdk.ErrObjectNotFound) {
+		_, err := client.Databases.ShowByID(ctx, id.DatabaseId())
+		if err != nil {
+			log.Printf("[DEBUG] database %s for schema %s not found", id.DatabaseId().Name(), id.Name())
 			d.SetId("")
-			return diag.Diagnostics{
-				diag.Diagnostic{
-					Severity: diag.Warning,
-					Summary:  "Failed to query secondary database. Marking the resource as removed.",
-					Detail:   fmt.Sprintf("DatabaseName: %s, Err: %s", id.FullyQualifiedName(), err),
-				},
+			return nil
```

Comment:
If we return here I think it would be better to also return Diag warning as in the case below to inform user about failed database query and that we're marking resource as removed. Otherwise, user will get meaningless `produced an unexpected new value: Root │ object was present, but now absent.` error.

Reply:
Done.